### PR TITLE
Migrate Mistral AI from Gson to Jackson

### PR DESCRIPTION
**Pull request by czelabueno**
## Issue
https://github.com/langchain4j/langchain4j/issues/1042

## Change
This PR migrates Mistral AI module only from Gson to Jackson according refactor plan.
I've decided to begin with JSON migration before implement more breaking changes in Mistral AI module.

@langchain4j take account that this PR is pre-requisite to implement Codestral fill-in-the-middle endpoint. [FIM feature](https://docs.mistral.ai/capabilities/code_generation/) will allow to the users define the starting point of the code using a `prompt`. FIM feature PR is waiting for this PR.

## General checklist

- [ ] There are no breaking changes
- [X] I have added unit and integration tests for my change
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [X] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)

### CODE REVIEW

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java [hunk 1]
```diff
@@ -169,16 +172,7 @@ private Response<AiMessage> generate(List<ChatMessage> messages,
         );
     }
 
-    public static MistralAiChatModelBuilder builder() {
-        for (MistralAiChatModelBuilderFactory factory : loadFactories(MistralAiChatModelBuilderFactory.class)) {
-            return factory.get();
```
Comment (langchain4j) [inline langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java]: is this removal intended?
same with all other classes below

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java [hunk 2]
```diff
@@ -59,7 +62,7 @@ public class MistralAiChatModel implements ChatLanguageModel {
      * @param maxRetries   the maximum number of retries for API requests. It uses the default value 3 if not specified
      */
     @Builder
-    public MistralAiChatModel(String baseUrl,
+    private MistralAiChatModel(String baseUrl,
```
Comment (langchain4j) [inline langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java]: why/is this change required?
same with all other classes below

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/internal/client/DefaultMistralAiClient.java [hunk 3]
```diff
@@ -132,11 +129,11 @@ public void onEvent(EventSource eventSource, String id, String type, String data
                     handler.onComplete(response);
                 } else {
                     try {
-                        MistralAiChatCompletionResponse chatCompletionResponse = GSON.fromJson(data, MistralAiChatCompletionResponse.class);
+                        MistralAiChatCompletionResponse chatCompletionResponse = OBJECT_MAPPER.readValue(data, MistralAiChatCompletionResponse.class);
                         MistralAiChatCompletionChoice choice = chatCompletionResponse.getChoices().get(0);
 
                         String chunk = choice.getDelta().getContent();
-                        if (isNotNullOrBlank(chunk)) {
+                        if (isNotNullOrEmpty(chunk)) {
```
Comment (langchain4j) [inline langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/internal/client/DefaultMistralAiClient.java]: worth extracting this bugfix into a separate PR?

**Review by ghost — COMMENTED** [review]
@czelabueno thanks a lot!

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java (hunk 1 above)
Comment (czelabueno) [reply langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java]: ohh I though ( I wrongly assumed it by mistake) that it was part of the refactoring plan  bcs the anthropic module haven't spi folder and Builder factories, if it is not, I can put back it. Do you want me to put back it? is it used by spring and quarkus extensions?

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java (hunk 2 above)
Comment (czelabueno) [reply langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java]: same comment. I saw that anthropic have it as private and I though that it's part of refactoring plan. If it's not I put back the builder factory as before

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/internal/client/DefaultMistralAiClient.java (hunk 3 above)
Comment (czelabueno) [reply langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/internal/client/DefaultMistralAiClient.java]: uhm maybe for tracking purpose. It's a trivial change but I can fix it in a separate PR for sure.

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java (hunk 1 above)
Comment (langchain4j) [reply langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java]: Right, Anthropic does not have `spi` directory (yet). Please put it back as it is used by Quarkus extension

File: langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java (hunk 2 above)
Comment (langchain4j) [reply langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModel.java]: I would drop this change for the moment to not break anything

**Comment by czelabueno** [comment]
@langchain4j all the comments were solved. Pls, check it again and then I'd create other PR for the bug and for new feature for `Codestral` FIM endpoint. Thanks!

**Review by ghost — APPROVED** [review]
@czelabueno thanks a lot!
